packages/framework/src/Component/Elasticsearch/IndexDefinition.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Component\Elasticsearch;
+
+use Shopsys\FrameworkBundle\Component\Elasticsearch\Exception\ElasticsearchIndexException;
+
+class IndexDefinition
+{
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\Elasticsearch\AbstractIndex
+     */
+    protected $index;
+
+    /**
+     * @var string
+     */
+    protected $definitionsDirectory;
+
+    /**
+     * @var string
+     */
+    protected $indexPrefix;
+
+    /**
+     * @var int
+     */
+    protected $domainId;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Elasticsearch\AbstractIndex $index
+     * @param string $definitionsDirectory
+     * @param string $indexPrefix
+     * @param int $domainId
+     */
+    public function __construct(AbstractIndex $index, string $definitionsDirectory, string $indexPrefix, int $domainId)
+    {
+        $this->index = $index;
+        $this->definitionsDirectory = $definitionsDirectory;
+        $this->indexPrefix = $indexPrefix;
+        $this->domainId = $domainId;
+    }
+
+    /**
+     * @return array
+     */
+    public function getDefinition(): array
+    {
+        $decodedDefinition = json_decode($this->getDefinitionFileContent(), true);
+        if ($decodedDefinition === null) {
+            throw ElasticsearchIndexException::invalidJsonInDefinitionFile(
+                $this->getIndex()->getName(),
+                $this->getDefinitionFilepath()
+            );
+        }
+
+        return $decodedDefinition;
+    }
+
+    /**
+     * @return string
+     */
+    protected function getDefinitionFilepath(): string
+    {
+        return $this->definitionsDirectory . $this->getIndex()->getName() . '/' . $this->getDomainId() . '.json';
```

Comment:
filename 1.json etc is very bad to searching.... Imagine all entities called `Entity` and only namespace says what is it...

It haas very bad DX... see #1286

Reply:
I udestand your trouble and agree with you. We are not planning to change it in this PR as it is not a part of this change. I have looked into the linked PR and I am missing any suggestion from you how it could be done better for you.

When we come up with a better solution it is not hard to change later.